lib/pure/strutils.nim

```diff
@@ -2319,6 +2319,46 @@ proc removeSuffix*(s: var string, suffix: string) {.
     newLen -= len(suffix)
     s.setLen(newLen)
 
+proc removePrefix*(s: var string, chars: set[char] = Newlines) {.
```

Comment:
IMHO, renaming this function to stripLeft will be the best name

Reply:
@data-man but that would basically be:

`strip(str, leading = false)` == `stripRight`
`strip(str, trailing = false)` == `stripLeft`

which both return the new string. These mutate the given string.

Reply:
Just more correctly (linguistically or grammatically)
A suffix (or prefix) is a word, that is, a letter or several letters in a certain order.
 "ab" != "ba", and this is correctly applied to words.
{'a', 'b'} == {'b', 'a'} but this not a words.

----
Sorry for using google translate :(

Reply:
My suggestions:
```Nim
strip(s: string; leading = true; trailing = true; chars: set[char] = Whitespace): string
removeSuffix(s: var string; chars: set[char] = Newlines)
```
must be deprecated.
And added their inplace variants:
```Nim
strip(s: string; chars: set[char] = Whitespace): string
stripLeft(s: string; chars: set[char] = Whitespace): string
stripRight(s: string; chars: set[char] = Whitespace): string

strip(var s: string; chars: set[char] = Whitespace)
stripLeft(var s: string; chars: set[char] = Whitespace)
stripRight(var s: string; chars: set[char] = Whitespace)
```

Reply:
I'm good with the general idea because I like more composable functions, and tbh that did trip me up when I first used `removeSuffix`. But I don't think overloads based on the param being `var` or not are common or desired, and also don't think the return type is factored into the proc resolution

Reply:
@citycide they are, all these procs would be valid and you would be able to use both strip variants

Reply:
I think the current consensus is that it's a bad idea to overload based on `var/let`. The convention of using f.ex `strip` for the in-place proc and `striped` for the other is used in some modules.

Reply:
@GULPF the overloading here is also based on the existence of return value, and Nim disallows implicitly discarded values by default

Reply:
@Yardanico @GULPF I came up in JS so my familiarity is with distinct `strip` / `stripped` functions. That said, I think in a language like Nim that:

1) supports proc overloading
2) has static types
3) forces explicit discards

the overload actually is more appealing. Definitely would like a final say on what's idiomatic but I'm down to rework these if that's called for.